src/xunit.execution/Sdk/Reflection/Reflector.cs

```diff
@@ -33,10 +34,13 @@ public static object[] ConvertArguments(object[] args, Type[] types)
 
             if (args.Length == types.Length)
             {
+                var converted = new List<object>(args.Length);
                 for (var idx = 0; idx < args.Length; idx++)
                 {
-                    args[idx] = ConvertArgument(args[idx], types[idx]);
+                    converted.Add(ConvertArgument(args[idx], types[idx]));
```

Comment:
I had to change this to avoid:

> System.ArrayTypeMismatchException : Attempted to access an element as a type incompatible with the array.

Reply:
args is defined as `object[]`, so this confuses me.

Reply:
That makes two of us 😕

Reply:
Was it failing in production, or only in the tests? It might be because your tests aren't passing an object[], but rather a string[] (implicitly typed).

Reply:
You hit the jackpot. I'll push fixups.